app/gui/view/graph-editor/src/new_node_position.rs

```diff
@@ -113,7 +127,42 @@ pub fn below_line_and_left_aligned(
     let y_offset = y_gap + node::HEIGHT / 2.0;
     let starting_point = Vector2(align_x, line_y - y_offset);
     let direction = Vector2(-1.0, 0.0);
-    on_ray(graph_editor, starting_point, direction).unwrap()
+    let pos = on_ray(graph_editor, starting_point, direction).unwrap();
+    // Perform Magnet Alignment if the node is pushed left to the free space.
+    if pos.x != align_x {
+        magnet_alignment_x(graph_editor, pos)
+    } else {
+        pos
+    }
+}
+
+/// Perform a Magnet Alignment algorithm both horizontally and vertically.
+///
+/// Magnet Alignment tries to place a node so that it is aligned to the positions of the other
+/// nodes, creating a "grid" of nodes.
+pub fn magnet_alignment_xy(graph_editor: &GraphEditorModel, position: Vector2) -> Vector2 {
+    magnet_alignment(graph_editor, position, false)
+}
+
+/// Perform a Magnet Alignment algorithm horizontally only.
+///
+/// Magnet Alignment tries to place a node so that it is aligned to the positions of the other
+/// nodes, creating a "grid" of nodes.
+pub fn magnet_alignment_x(graph_editor: &GraphEditorModel, position: Vector2) -> Vector2 {
+    magnet_alignment(graph_editor, position, true)
+}
+
+fn magnet_alignment(
+    graph_editor: &GraphEditorModel,
+    position: Vector2,
+    only_horizontally: bool,
```

Comment:
Instead of passing bool as argument, we could make an enum with explanatory names. Then we could have only one public `magnet_alginment` method instead of three (there will be no need for _x/_xy variants if enum names will be self-explanatory).

Reply:
Done!